app/router.js

```diff
@@ -25,17 +50,25 @@ const Router = Ember.Router.extend({
 
 Router.map(function() {
     this.route('page-not-found', {path: '/*bad_url'});
-    this.route('index', {path: 'preprints'});
-    this.route('page-not-found', {path: 'preprints/page-not-found'});
-    this.route('submit', {path: 'preprints/submit'});
-    this.route('discover', {path: 'preprints/discover'});
-    this.route('content', {path: '/:preprint_id' });
-    this.route('provider', {path: 'preprints/:slug'}, function() {
-        this.route('content', {path: '/:preprint_id'});
-        this.route('discover');
+
+    if (provider) {
```

Comment:
Since changing the provider requires a full page reload, the entire app, including the router, will be evaluated each time.